Sources/AST/Statement/DoCatchStatement.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  DoCatchStatement.swift
+//  AST
+//
+//  Created by Ethan on 05/11/2018.
+//
+import Source
+import Lexer
+
+/// A do catch block.
+public struct DoCatchStatement: ASTNode {
+  // TODO(ethan): Once we have exceptions, change this error to have type `Exceptions`
+  public var error: Expression
+  public var doBody: [Statement]
+  public var catchBody: [Statement]
```

Comment:
Nit: it would be better if you put these in the same order as you have in the constructor. 
Additionally do these need to be mutable necessarily?